pkg/controller/reconciler/deployment_reconciler.go

```diff
@@ -546,6 +560,14 @@ func (r *DeploymentReconciler) updateDeployment(ctx context.Context, deployment
 		}
 	}
 
+	// Add the hash of the secret data to the Pod definition. This will force a rollout when the secrets
+	// change.
+	hash := kubernetes.HashSecretData(secret.Data)
+	if deployment.Spec.Template.ObjectMeta.Annotations == nil {
+		deployment.Spec.Template.ObjectMeta.Annotations = map[string]string{}
+	}
+	deployment.Spec.Template.ObjectMeta.Annotations[kubernetes.AnnotationSecretHash] = hash
```

Comment:
This makes sure we rollout new pods when the connections or connection data changes